src/res_tracker.cc

```diff
@@ -23,7 +23,8 @@ void ResTracker::Create(Agent* creator) {
 
   parent1_ = 0;
   parent2_ = 0;
-  Record();
+  bool no_bump = true;
+  Record(no_bump);
```

Comment:
Just for my education.... this change isn't strictly related to the purpose of this PR.  It seems like we previously bumped the ID unnecessarily here, but probably decided that was fine because it only happened once per resource and it was deemed too cumbersome to avoid it. Now that we have a good reason to avoid it in some cases and have added that capability, you've gone ahead and introduced it here, too.

Do you think that's a good characterization? (recognizing that we'd both need to speculate on our past intent...)

Reply:
Yes, but also we create new resources a lot in packaging. So the old workflow of create new resource (comes with new state id) then record new resource (bump state ID, second line on resource table for the exact same resource we just created) would just be much more obvious because it would happen much, much more

Reply:
Oh! You raised something important that I missed though. This is important when a new resource is created via `Material::Create`, which calls `ResTracker::Create`, but not when a resource is created via `new Material`. Right now I use `new Material`, because that's what `ExtractComp` uses. But maybe that's not great? Am I wrong in thinking that using `Material::Create` is better practice?

The only issue is that a resource, once created, does not know its creator any more. It doesn't save that information. So if I'm trying to create a new resource from within a resource, I can't actually call `Material::Create`, I don't have enough info. So in order to use `Create` from within `PackageExtract`, I think I'd have to pass the creator in to `Resource::Package`, and then down to `PackageExtract` ? Does this make sense? or is using `new` perfectly fine anyway?

Reply:
I guess as long as you agree that it's not bad that `ExtractComp` currently uses `new Material` instead of `Create`, we should probably keep packaging out of `Create` anyway. Because I don't think new materials created to be packaged should end up in the `ResCreators` table since they're not being created from thin air (as with Sources) they're being created using quantity from existing resources